tap_github/repository_streams.py

```diff
@@ -1907,6 +1909,293 @@ def query(self) -> str:
     ).to_dict()
 
 
+class DiscussionsStream(GitHubGraphqlStream):
+    """Defines stream fetching discussions from each repository."""
+
+    name = "discussions"
+    query_jsonpath = "$.data.repository.discussions.nodes.[*]"
+    primary_keys: ClassVar[list[str]] = ["id"]
+    replication_key = "updated_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys: ClassVar[list[str]] = ["repo", "org"]
```

Comment:
It looks like there's a best practice of using `repo_id` here instead of the repo and org names ([ref 1](https://github.com/MeltanoLabs/tap-github/issues/122), [ref 2](https://github.com/MeltanoLabs/tap-github/issues/149))!